call-home.sh

```diff
@@ -0,0 +1,326 @@
+        usage 1
+    fi
+
+    if [[ -z "$PERCONA_DEPLOYMENT_METHOD" ]];
+    then
+        printf "PERCONA_DEPLOYMENT_METHOD is not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+
+    if [[ -z "$PERCONA_INSTANCE_ID" ]];
+    then
+        printf "PERCONA_INSTANCE_ID is not provided. %s\n" "$USAGE_TEXT" >&2
+        usage 1
+    fi
+}
+
+read_telemetry_config_file()
+{
+    while IFS=":" read -r key value;
+    do
+        # Trim possible leading and trailing whitespaces
+        key=$(echo $key | xargs)
+        value=$(echo $value | xargs)
+        telemetry_config_map[$key]=$value
+    done < "$PERCONA_TELEMETRY_CONFIG_FILE_PATH"    
+}
+
+# Creates the instance id file if one doesn't exist
+# If instance id file exists, we already reported this instance and there is no more work to do
+check_or_create_percona_instance_id_file()
+{
+    local should_create=0
+
+    if [[ ! -f "$PERCONA_TELEMETRY_CONFIG_FILE_PATH" ]];
+    then
+        # There is no file at all
+        should_create=1
+    elif [[ $(grep -c "^\s*instanceId\s*:\s*[0-9a-fA-F]\{8\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{4\}-[0-9a-fA-F]\{12\}\s*$" $PERCONA_TELEMETRY_CONFIG_FILE_PATH) -ne 1 ]];
+    then
+        # There is a file, but does not contain valid instanceId (spoiled?)
+        should_create=1
+    fi
+
+    if [[ $should_create -eq 1 ]];
+    then
+        # If instance ID is not provided externally, generate it
+        if [[ -z "$PERCONA_INSTANCE_ID" ]];
+        then
+            PERCONA_INSTANCE_ID=$(cat /proc/sys/kernel/random/uuid)
+        fi
+
+        # Make sure the storage directory exists
+        mkdir -p $(dirname $PERCONA_TELEMETRY_CONFIG_FILE_PATH)
+
+        # Truncate the file even if it existed, but was spoiled
+        printf "%s:%s\n" "instanceId" $PERCONA_INSTANCE_ID > $PERCONA_TELEMETRY_CONFIG_FILE_PATH
+    else
+        # Restore InstanceId and the whole context from the file
+        read_telemetry_config_file
+        PERCONA_INSTANCE_ID=${telemetry_config_map["instanceId"]}
+    fi
+
+    # Check if this product was already reported
+    if [[ ! -z ${telemetry_config_map[$PERCONA_PRODUCT_FAMILY]} ]];
+    then
+        # This product already has been reported, exit script
+        # echo "Product already reported. Exiting"
+        exit_script
+    fi
+}
+ 
+
+mark_product_as_reported()
+{
+   # This product hasn't been reported, so mark it as reported by appending ths product
+    printf "%s:%s\n" $PERCONA_PRODUCT_FAMILY "1" >> $PERCONA_TELEMETRY_CONFIG_FILE_PATH
+}
+
+collect_data_for_report()
+{
+    json_message_map["id"]=$(cat /proc/sys/kernel/random/uuid)
+    json_message_map["createTime"]="$(date --iso-8601='seconds')"
+    json_message_map["instanceId"]=$PERCONA_INSTANCE_ID
+    json_message_map["product_family"]=$PERCONA_PRODUCT_FAMILY
+
+    metrics_map["pillar_version"]=$PERCONA_PRODUCT_VERSION
+    metrics_map["OS"]=${PERCONA_OPERATING_SYSTEM}
```

Comment:
In the list of basic metrics for phase 0 there is HW architecture (at least from https://www.notion.so/percona/Pillars-product-usage-tracking-3a933f1e51404cd68e6e40691912556f ):
- operating system name and version (uname -ro or equivalent)
*** - hardware architecture (uname -m equivalent)
- pillar software installed
- pillar software version including hotfix number

Will we include it to OS?

Reply:
Good point! Added as `hardware_arch`